Automattic/wp-calypso
# Gutenberg: Add an action for setting a site's selected editor.

This PR adds the `EDITOR_TYPE_SET` action that optimistically adds/updates `state.selectedEditor[:siteId]` for a given site, with a value of either `gutenberg` or `classic`.

The data layer handles calling the Gutenberg opt-in endpoint from D19412-code to update the server-side. Since we're still honoring the user's choice of editor even if the API call fails, there's no need to handle errors (and no need for an error notice, it would just add confusion for the user).

Since this is easier to test if something is importing the selector (and therefore the data layer handler), I've gone ahead and used the new selector to wire up the opt-in button in the dialog.

<img width="1196" alt="47056879-34ad0a80-d173-11e8-9672-6b1c4087f00d" src="https://user-images.githubusercontent.com/349751/47382555-6a904880-d6b7-11e8-88e7-6201939501be.png">

This is part of work required for #27431 .

**Testing instructions**
* Load `/post` and choose a test site.
* Click the opt-in visual in the editor sidebar, then choose "Try the new editor".
* Verify an API call is made to `/sites/:site/gutenberg?editor=gutenberg`
* Verify you are redirected to `/gutenberg/post/:site`.
* Verify that site now has the `enable-gutenberg` sticker (if it didn't already).
* From the console, dispatch the action again, this time setting classic: `dispatch({type:'EDITOR_TYPE_SET',siteId:112391175, editor:'classic'})`
* Again, verify the API call is made, and the site no longer has the `enable-gutenberg` blog sticker.
